src/org/labkey/test/tests/DataClassFolderExportImportTest.java

```diff
@@ -0,0 +1,351 @@
+        }
+
+        // now ensure expected data in the sampleType made it to the destination folder
+        for (Map exportedRow : sourceRowData)
+        {
+            // find the map from the exported project with the same name
+            Map<String, String> matchingMap = destRowData.stream().filter(a-> a.get("Name").equals(exportedRow.get("Name")))
+                    .findFirst().orElse(null);
+            assertNotNull("expect all matching rows to come through", matchingMap);
+            assertThat("expect export and import values to be equivalent",
+                    exportedRow.get("intColumn"), equalTo(matchingMap.get("intColumn")));
+            assertThat("expect export and import values to be equivalent",
+                    exportedRow.get("stringColumn"), equalTo(matchingMap.get("stringColumn")));
+            assertThat("expect export and import values to be equivalent",
+                    exportedRow.get("decimalColumn"), equalTo(matchingMap.get("decimalColumn")));
+            assertThat("expect export and import values to be equivalent",
+                    exportedRow.get("attachmentColumn"), equalTo(matchingMap.get("attachmentColumn")));
+        }
+    }
+
+    @Test
+    public void testExportImportMissingValueDataClass() throws Exception
+    {
+        final String MV_INDICATOR_01 = "Q";
+        final String MV_DESCRIPTION_01 = "Data currently under quality control review.";
+        final String MV_INDICATOR_02 = "N";
+        final String MV_DESCRIPTION_02 = "Required field marked by site as 'data not available'.";
+        final String MV_INDICATOR_03 = "X";
+        final String MV_DESCRIPTION_03 = "Here is a non system one.";
+
+        List<Map<String, String>> missingValueIndicators = new ArrayList<>();
+        missingValueIndicators.add(Map.of("indicator", MV_INDICATOR_01, "description", MV_DESCRIPTION_01));
+        missingValueIndicators.add(Map.of("indicator", MV_INDICATOR_02, "description", MV_DESCRIPTION_02));
+        missingValueIndicators.add(Map.of("indicator", MV_INDICATOR_03, "description", MV_DESCRIPTION_03));
+
+        final String REQUIRED_FIELD_NAME = "requiredField";
+        final String MISSING_FIELD_NAME = "missingField";
+        final String INDICATOR_FIELD_NAME = MISSING_FIELD_NAME + "MVIndicator";
+
+        String subfolder = "missingValueDataClassExportFolder";
+        String subfolderPath = getProjectName() + "/" + subfolder;
+        String testDataClass = "missing?Value*Data//Class";     // having the dataClass with non-file-legal chars in it is intentional,
+                                                                // the import/export processes will write temp-files with names derived from
+                                                                // the dataClass name.  This ensures that the sanitized name on export can
+                                                                // be successfully matched up with the intended dataClass on import
```

Comment:
This looks like a good thing to test for, would it make sense to put this in its own small test case/function? If the export fails then the test does not end up validating missing values (which the title says it will validate).

Reply:
I'm not sure that it will.  I'm using non-file-char dataclass names in both scenarios to provide this coverage for dataclasses with different field types, including attachments (which I suspect are central to this scenario).